File: android/app/src/main/kotlin/com/example/test_twilio/arguments/BasicChatClientArgument.kt
package com.example.test_twilio.arguments

class BasicChatClientArgument(var accessToken: String?, var firebaseToken: String?) {

    companion object {
        fun fromMap(data: HashMap<String, Any>): BasicChatClientArgument {
            val accessToken = data.get("accessToken")?.toString()
            val firebaseToken = data.get("firebaseToken")?.toString()
            return BasicChatClientArgument(accessToken = accessToken, firebaseToken = firebaseToken);
        }
    }
}

File: lib/services/arguments/basic_chat_client_argument.dart
class BasicChatClientArgument {
  String? accessToken;
  String? firebaseToken;

  BasicChatClientArgument(this.accessToken, this.firebaseToken);

  Map<String, dynamic> toMap() {
    return {"accessToken": accessToken, "firebaseToken": firebaseToken};
  }
}